pkg/controllers/user/pipeline/controller/clusterpipeline/clusterpipelinesyncer.go

```diff
@@ -0,0 +1,108 @@
+package clusterpipeline
+
+import (
+	"github.com/pkg/errors"
+	"github.com/rancher/rancher/pkg/controllers/user/pipeline/utils"
+	"github.com/rancher/types/apis/core/v1"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Syncer struct {
+	cluster    *config.UserContext
```

Comment:
@zionwu @gitlawr I'd prefer seeing the *Lister, *Interface fields instead of top level cluster as a field. This way we can clearly see what apis are being called, and also see if the call to api is made directly vs fetching info from cache.

Reply:
Sure, I will change in this way